### PR TITLE
added Getting started section to README

### PR DESCRIPTION
**Pull request by barabo**
### Checklist - did you ...

- [ ] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [ ] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

**Comment by dholth** [comment]
Looking good

**Review by dholth — APPROVED** [review]
LGTM. Are we including README.md in the sphinx docs?

**Comment by barabo** [comment]
> LGTM. Are we including README.md in the sphinx docs?

I haven't done anything to add the README to sphinx - I looked into it and wasn't sure how.  Happy to add it if you can give me a pointer.

**Comment by barabo** [comment]
Something about the changes in this branch seem to introduduce failures in two unit tests that pass against `main`.  :-\

<details><summary>Click for details...</summary>

```python
____________________________________________________________________________________________ test_current_index_reduces_space ____________________________________________________________________________________________

index_data = PosixPath('/tmp/pytest-of-codespace/pytest-7/test_current_index_reduces_spa0/index_data')

    @pytest.mark.skipif(
        not hasattr(context, "use_only_tar_bz2") or getattr(context, "use_only_tar_bz2"),
        reason="conda is set to auto-disable .conda for old conda-build.",
    )
    def test_current_index_reduces_space(index_data):
        repodata = Path(index_data, "time_cut", "repodata.json")
        with open(repodata) as f:
            repodata = json.load(f)
        assert len(repodata["packages"]) == 7
        assert len(repodata["packages.conda"]) == 3
        trimmed_repodata = conda_index.index._build_current_repodata(
            "linux-64", repodata, None
        )
    
        tar_bz2_keys = {
            "two-because-satisfiability-1.2.11-h7b6447c_3.tar.bz2",
            "two-because-satisfiability-1.2.10-h7b6447c_3.tar.bz2",
            "depends-on-older-1.2.10-h7b6447c_3.tar.bz2",
            "ancient-package-1.2.10-h7b6447c_3.tar.bz2",
            "one-gets-filtered-1.3.10-h7b6447c_3.tar.bz2",
        }
        # conda 4.7 removes .tar.bz2 files in favor of .conda files
        if conda_47:
            tar_bz2_keys.remove("one-gets-filtered-1.3.10-h7b6447c_3.tar.bz2")
    
        # .conda files will replace .tar.bz2 files.  Older packages that are necessary for satisfiability will remain
        assert set(trimmed_repodata["packages"].keys()) == tar_bz2_keys
        if conda_47:
>           assert set(trimmed_repodata["packages.conda"].keys()) == {
                "one-gets-filtered-1.3.10-h7b6447c_3.conda"
            }
E           AssertionError: assert {'one-gets-filtered-1.2.10-h7b6447c_3.conda', 'one-gets-filtered-1.2.11-h7b6447c_3.conda', 'one-gets-filtered-1.3.10-h7b6447c_3.conda'} == {'one-gets-filtered-1.3.10-h7b6447c_3.conda'}
E             Extra items in the left set:
E             'one-gets-filtered-1.2.10-h7b6447c_3.conda'
E             'one-gets-filtered-1.2.11-h7b6447c_3.conda'
E             Full diff:
E               {
E             +  'one-gets-filtered-1.2.10-h7b6447c_3.conda',
E             +  'one-gets-filtered-1.2.11-h7b6447c_3.conda',
E                'one-gets-filtered-1.3.10-h7b6447c_3.conda',
E               }

tests/test_index.py:975: AssertionError
__________________________________________________________________________________ test_current_index_version_keys_keep_older_packages ___________________________________________________________________________________

index_data = PosixPath('/tmp/pytest-of-codespace/pytest-7/test_current_index_version_key0/index_data')

    def test_current_index_version_keys_keep_older_packages(index_data):
        pkg_dir = Path(index_data, "packages")
    
        # pass no version file
        conda_index.api.update_index(pkg_dir)
        with open(os.path.join(pkg_dir, "osx-64", "current_repodata.json")) as f:
            repodata = json.load(f)
        # only the newest version is kept
>       assert len(repodata["packages"]) == 1
E       AssertionError: assert 2 == 1
E        +  where 2 = len({'dummy-package-1.0-0.tar.bz2': {'build': '0', 'build_number': 0, 'depends': ['python 2.7.*'], 'md5': 'fe3ca1adcd2b7799a5eba03cc774725d', ...}, 'dummy-package-2.0-0.tar.bz2': {'build': '0', 'build_number': 0, 'depends': ['python 2.7.*'], 'md5': '621277e18c2627643b156a0f3fad3082', ...}})

tests/test_index.py:1002: AssertionError
-------------------------------------------------------------------------------------------------- Captured stderr call --------------------------------------------------------------------------------------------------
```

</details>

**Review by kathatherine — COMMENTED** [review]
I'd say this is definitely a moderately advanced getting started README, but as long as you expect your users to not be totally newbies, it likely works. I left a few comments on being more explicit about a few things.

File: README.md [hunk 1]
```diff
@@ -1,14 +1,113 @@
...
+├── channeldata.json
+├── index.html
+├── noarch
+│   ├── current_repodata.json
+│   ├── index.html
+│   ├── repodata_from_packages.json
+│   └── repodata.json
+└── rss.xml
+
+(base) @barabo ➜ /workspaces/conda-index (main) $ sqlite3 ./noarch/.cache/cache.db 'select * from stat where stage = "fs" limit 10;'
```
[the remark this answers is not in the document]
Comment (kathatherine) [reply README.md:89]: That's good to know. Then maybe an addition of "You can use `sqlite`, which can be installed with conda, to view your cache database." to my suggestion above.

**Pull request closed without merging**